### PR TITLE
httpbakery/agent: support authinfo in environment variables

### PR DESCRIPTION
**Pull request by rogpeppe**
This makes it easy to get agent authinfo from environment
variables instead of a file.

### CODE REVIEW

File: httpbakery/agent/agent.go [hunk 1]
```diff
@@ -40,6 +41,63 @@ type Agent struct {
...
+// BAKERY_PRIVATE_KEY - the private key of the agent, base64 encoded.
+// BAKERY_PUBLIC_KEY - the public key of the agent, base64 encoded.
+// BAKERY_USERS - space-separated (username, URL) pairs of username
+// and identity manager URLs.
+//
+// For example:
+//
+// BAKERY_PUBLIC_KEY='wg/rUMm5cy9O00TaS9pgDIHnFMwhrisI0KDSg4+KkGM='
+// BAKERY_PRIVATE_KEY='S529MFrVUWB/VqcsPABJSAlnY/57fdHi4cL9HdK9ZQw='
+// BAKERY_USERS='https://api.jujucharms.com/identity bob https://api.staging.jujucharms.com/identity'
```
Comment (mhilton) [inline httpbakery/agent/agent.go]: This is significantly more complex than I had in mind. I was essentially looking for a fix to https://github.com/juju/charmstore/issues/776. This is not usable that simply. We need to be able to provide API access to jujucharms.com through a mechanism that it "set these two environment variables", "use the tools" type system. The same as people expect from AWS. I'm not interested that we have this crazy bakery setup inside the API that complexity should not be forced onto out users.

File: httpbakery/agent/agent_test.go [hunk 2]
```diff
@@ -117,6 +121,36 @@ func (s *agentSuite) TestSetUpAuth(c *gc.C) {
...
+	_, err = f.Write(data)
+	c.Assert(err, gc.Equals, nil)
+	f.Close()
+
+	os.Setenv("BAKERY_AGENT_FILE", f.Name())
+
+	authInfo1, err := agent.AuthInfoFromEnvironment()
+	c.Assert(err, gc.Equals, nil)
+	c.Assert(authInfo1, jc.DeepEquals, authInfo)
+}
```
Comment (mhilton) [inline httpbakery/agent/agent_test.go:152]: Please also add a test for the error when the environment variable isn't set.

**Review by mhilton — APPROVED** [review]
LGTM modulo an additional test.

File: httpbakery/agent/agent_test.go (hunk 2 above)
Comment (rogpeppe) [reply httpbakery/agent/agent_test.go:152]: done.

**Pull request merged**